src/mococrw/symmetric_crypto.h

```diff
@@ -193,6 +193,17 @@ class AuthenticatedEncryptionI
      */
     virtual void setAuthTag(const std::vector<uint8_t> &tag) = 0;
 
+    /**
+     * Set associated data.
+     *
+     * This function sets the associated data in AEAD modes for authenticating while encrypting or verifying while
+     * decrypting. setAssociatedData() must be called before finalizing decryption using SymmetricCipherI::finish()
+     * and before putting encrypted data to the message using SymmetricCipherI::update()
+     *
+     * @param associatedData chunk of data to associate/verify.
+     */
+    virtual void setAssociatedData(const std::vector<uint8_t> &associatedData) = 0;
```

Comment:
If we want ABI stability, we probably need to add new functions to the end?

Reply:
I don't know what the prerequisites for ABI stability with our configuration are so I'm just trusting you and moving it to the end of the class declaration.